virtualmachine/vsphere/util.go

```diff
@@ -1107,8 +1160,8 @@ var getVMLocation = func(vm *VM, dcMo *mo.Datacenter) (l location, err error) {
 				err = fmt.Errorf("No suitable hosts found in the cluster")
 				return
 			}
-			n := util.Random(1, len(filteredHosts))
-			l.Host = filteredHosts[n-1]
+			n := util.Random(1, len(filteredHosts)-1)
```

Comment:
can we use ChooseRandomString() here as well?

Reply:
the filteredHosts is not string actually, it is a govmomi object. The function ChooseRandomString should be generalised to handle such request. Will make these changes in the next PR.